bouncer/src/repo/core/handler/repo_database_handler_mongo.cpp

```diff
@@ -701,157 +424,109 @@ bool MongoDatabaseHandler::insertDocument(
-		{
-			std::string errString(e.what());
-			errMsg += errString;
-		}
-	}
-	else
-	{
-		errMsg = "Unable to insert Document, database(value : " + database + ")/collection(value : " + collection + ") name was not specified";
-	}
-
-	return success;
+	auto client = clientPool->acquire();
+	auto db = client->database(database);
+	auto col = db.collection(collection);
+	col.insert_one(obj.view());
 }
 
-bool MongoDatabaseHandler::insertManyDocuments(
+void MongoDatabaseHandler::insertManyDocuments(
 	const std::string &database,
 	const std::string &collection,
 	const std::vector<repo::core::model::RepoBSON> &objs,
-	std::string &errMsg,
 	const Metadata& binaryStorageMetadata)
 {
-	bool success = false;
-
-	if (!database.empty() || collection.empty())
-	{
-		try {
-			auto fileManager = fileservice::FileManager::getManager();
-
-			fileservice::BlobFilesHandler blobHandler(fileManager, database, collection, binaryStorageMetadata);
-
-			for (int i = 0; i < objs.size(); i += MAX_PARALLEL_BSON) {
-				std::vector<repo::core::model::RepoBSON>::const_iterator it = objs.begin() + i;
-				std::vector<repo::core::model::RepoBSON>::const_iterator last = i + MAX_PARALLEL_BSON >= objs.size() ? objs.end() : it + MAX_PARALLEL_BSON;
-				std::vector<mongo::BSONObj> toCommit;
-				do {
-					auto node = *it;
-					auto data = node.getBinariesAsBuffer();
-					if (data.second.size()) {
-						auto ref = blobHandler.insertBinary(data.second);
-						node.replaceBinaryWithReference(ref.serialise(), data.first);
-					}
-					toCommit.push_back(node);
-				} while (++it != last);
-
-				repoInfo << "Inserting " << toCommit.size() << " documents...";
-				worker->insert(getNamespace(database, collection), toCommit);
+	auto client = clientPool->acquire();
+	auto db = client->database(database);
+	auto col = db.collection(collection);
+
+	fileservice::BlobFilesHandler blobHandler(fileManager, database, collection, binaryStorageMetadata);
+
+	for (int i = 0; i < objs.size(); i += MAX_PARALLEL_BSON) {
+		std::vector<repo::core::model::RepoBSON>::const_iterator it = objs.begin() + i;
+		std::vector<repo::core::model::RepoBSON>::const_iterator last = i + MAX_PARALLEL_BSON >= objs.size() ? objs.end() : it + MAX_PARALLEL_BSON;
+		std::vector<bsoncxx::document::value> toCommit;
+		do {
+			auto node = *it;
+			auto data = node.getBinariesAsBuffer();
+			if (data.second.size()) {
+				auto ref = blobHandler.insertBinary(data.second);
+				node.replaceBinaryWithReference(ref.serialise(), data.first);
 			}
+			toCommit.push_back(node);
+		} while (++it != last);
 
-			blobHandler.finished();
-
-			success = true;
-		}
-
-		catch (mongo::DBException &e)
-		{
-			std::string errString(e.what());
-			errMsg += errString;
-		}
-	}
-	else
-	{
-		errMsg = "Unable to insert Document, database(value : " + database + ")/collection(value : " + collection + ") name was not specified";
+		repoInfo << "Inserting " << toCommit.size() << " documents...";
+		col.insert_many(toCommit);
```

Comment:
Yes the driver still throws (in fact it seems it's its only way of providing feedback!)

When I was looking at the callers, it appears all the operations are effectively "required" in the sense that a failure of any of them would leave the, e.g. scene, in an incomplete state. Therefore, the best thing to do in encountering an unexpected failure is to bail out, like before, but this way through letting the exception bubble up to the client where it can be printed to the log with all the information in the original exception object.

(The exception being where we catch the exception object in order to rethrow with additional context, such as bson field names during parsing.)